tests/test_general.py

```diff
@@ -21,6 +21,15 @@ def _foo():
     return _foo
 
 
+@pytest.fixture
```

Comment:
this is the test; should not be a fixture

Reply:
I'm a bit confused about this. Should I modify it to:

```
def func_for_ignore_exception_raise_final_exception():
    @ignore_exceptions(retry=1, retry_interval=1, raise_final_exception=True)
    def _foo():
        raise ValueError()

    with pytest.raises(ValueError):
         _foo()
```

Also, would you like me to modify the other functions to not be fixtures?

Reply:
define a function and decorate it.
fixtures are used as test setup and teardown; in this case calling the function is the actual test.